### DISCUSSION TITLE
Schema validation for unsigned types

### DISCUSSION THREAD

**Discussion by beikov** [hibernate/hibernate-orm#6584] [Polls]
The issue [HHH-16578](https://hibernate.atlassian.net/browse/HHH-16578) exposed a problem with schema validation that we need to decide on how to handle.

If the database has a column `tinyint unsigned`, what should happen in schema validation for an attribute of type `int` that has a column definition of `tinyint unsigned default 0`? The JDBC driver reports `Types.TINYINT` for the column type, and the type code we assume for the model is `Types.INTEGER`.

**Comment by sebersole** [comment]
I'm inclined to say this should fail for the reason you say - its possible to have overflow.

Though I do wonder about applications that simply never push "large" values to that column.  But if they are requesting validation, maybe its just best to tell them..

**Reply by jrenaat** [reply]
That was also Gavin's take; 'if they ask us to validate, let's validate'

**Comment by jrenaat** [comment]
I should perhaps give some context:
the user indeed maps an int to a tinyint column, however he has a columndefinition on that field matching the definition in the db (i.e. tinyint unsigned default '0').
As such, the validation fails. The issue being that if the columndefinition is reduced to just tinyint, the validation passes, which would appear to be inconsistent.

**Reply by beikov** [reply]
I just updated the text of the poll with that information :)

**Comment by jrenaat** [comment]
I'm inclined to think that it should fail, whether the columndefinition states "tinyint" or "tinyint ... ", forcing the use of @JdbcTypeCode for it to pass

**Comment by beikov** [comment]
After digging a bit deeper into this, it appears that there is simply a bug somewhere else. See https://github.com/hibernate/hibernate-orm/pull/6548/files#r1196727159 for details. Let's keep that discussion on the PR though. I'll close the poll :)
